packages/regl-worldview/src/commands/Command.js

```diff
@@ -9,45 +9,45 @@
 
 import * as React from "react";
 
-import type { ComponentMouseHandler, MouseEventEnum, RawCommand, Vec4, Color, Ray } from "../types";
-import { getIdFromColor, intToRGB } from "../utils/commandUtils";
+import type {
+  ComponentMouseHandler,
+  GetHitmap,
+  MouseEventEnum,
+  RawCommand,
+  Color,
+  Ray,
+  MouseEventObject,
+} from "../types";
 import { getNodeEnv } from "../utils/common";
 import { type WorldviewContextType } from "../WorldviewContext";
 import WorldviewReactContext from "../WorldviewReactContext";
 
 export const SUPPORTED_MOUSE_EVENTS = ["onClick", "onMouseUp", "onMouseMove", "onMouseDown", "onDoubleClick"];
 
-export type HitmapProp<T> = T & ({ colors: Vec4[] } | { color: Vec4 });
-export type GetObjectFromHitmapId<T> = (objectId: number, hitmapProps: HitmapProps<T>[]) => ?HitmapProps<T>;
-export type GetHitmapProps<T> = (children: T[]) => ?HitmapProp<T>;
 export type MarkerDefault = {
   id?: number,
   points?: Point[],
   color?: Color,
 };
-export type HitmapMarkerDefault = {
-  id?: number,
-  points?: Point[],
-  color?: Vec4,
+
+export type OptionalCommandProps = {
+  [MouseEventEnum]: ComponentMouseHandler,
+  layerIndex?: number,
+  getHitmap?: GetHitmap,
 };
 
-export type Props<T> = {
+type Props<T> = {
   [MouseEventEnum]: ComponentMouseHandler,
-  children?: T[],
-  getObjectFromHitmapId?: GetObjectFromHitmapId<HitmapProp<T>>,
-  hitmapProps?: HitmapProp<T>,
+  children: T[],
   layerIndex?: number,
   reglCommand: RawCommand<T>,
+  // Interactivity
+  getHitmap?: GetHitmap,
```

Comment:
Should use `...OptionalCommandProps` here. And maybe call those `CommonProps` or something?